core/chains/evm/txmgr/orm.go

```diff
@@ -228,10 +282,17 @@ func (o *orm) InsertEthTxAttempt(attempt *EthTxAttempt) error {
 }
 
 func (o *orm) InsertEthReceipt(receipt *EvmReceipt) error {
+	// convert to database representation
+	r := dbReceipt(*receipt)
+
 	const insertEthReceiptSQL = `INSERT INTO eth_receipts (tx_hash, block_hash, block_number, transaction_index, receipt, created_at) VALUES (
 :tx_hash, :block_hash, :block_number, :transaction_index, :receipt, NOW()
 ) RETURNING *`
-	err := o.q.GetNamed(insertEthReceiptSQL, receipt, receipt)
+	err := o.q.GetNamed(insertEthReceiptSQL, &r, &r)
+
+	// method expects original (destination) receipt struct to be updated
```

Comment:
If 1 struct deviates in future, how can we detect it?
Does it cause compile time failure?

Reply:
Yup!